File: src/main/java/com/sfh/pokeRogueBot/model/modifier/impl/AttackTypeBoosterModifierItem.java
package com.sfh.pokeRogueBot.model.modifier.impl;

import com.sfh.pokeRogueBot.model.enums.PokeType;
import com.sfh.pokeRogueBot.model.modifier.GeneratedPersistentModifierType;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class AttackTypeBoosterModifierItem extends PokemonHeldItemModifierItem implements GeneratedPersistentModifierType {

    public static final String TARGET = "AttackTypeBoosterModifierType";

    private PokeType moveType;
    private int boostPercent;

}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/modifier/impl/AttackTypeBoosterModifierItem.kt
package com.sfh.pokeRogueBot.model.modifier.impl

import com.sfh.pokeRogueBot.model.enums.PokeType
import com.sfh.pokeRogueBot.model.modifier.GeneratedPersistentModifierType

class AttackTypeBoosterModifierItem : PokemonHeldItemModifierItem(), GeneratedPersistentModifierType {

    companion object {
        const val TARGET = "AttackTypeBoosterModifierType"
    }

    var moveType: PokeType? = null
    var boostPercent: Int = 0
}